apps/mark-scan/frontend/src/pages/remove_invalidated_ballot_page.tsx

```diff
@@ -1,35 +1,44 @@
-import { useContext } from 'react';
-
 import { Button, P } from '@votingworks/ui';
+import { useHistory } from 'react-router-dom';
+import { useEffect } from 'react';
 import { confirmInvalidateBallot } from '../api';
-import { BallotContext } from '../contexts/ballot_context';
 import { CenteredPageLayout } from '../components/centered_page_layout';
 
-export function RemoveInvalidatedBallotPage(): JSX.Element {
-  const { resetBallot, endVoterSession } = useContext(BallotContext);
+interface Props {
+  paperPresent: boolean;
+}
+
+export function RemoveInvalidatedBallotPage(props: Props): JSX.Element {
+  const { paperPresent } = props;
 
   const confirmInvalidateBallotMutation = confirmInvalidateBallot.useMutation();
 
-  async function onPressContinue() {
-    // Reset session and ballot before changing the state machine state. If done
-    // in the reverse order the screen will flicker
-    await endVoterSession();
-    resetBallot();
+  const history = useHistory();
+  useEffect(() => {
+    history.push('/ready-to-review');
+  });
```

Comment:
As discussed before, not ideal to use react-router routing here. Should be simplified in https://github.com/votingworks/vxsuite/issues/3985